int8[] edges
float32[] weight_matrix
---
bool rec
